### PR TITLE
refactor: remove use of k8s secret

### PR DESCRIPTION
**Pull request by phillebaba**
## Description

Removes use of k8s secret functions.

## Related Issue

Relates to #2507

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/.github/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

**Review by AustinAbro321 — CHANGES_REQUESTED** [review]
K8s package is almost gone!

File: src/pkg/packager/remove.go [hunk 1]
```diff
@@ -107,17 +109,49 @@ func (p *Packager) Remove(ctx context.Context) (err error) {
...
+			if err != nil && !kerrors.IsNotFound(err) {
+				return err
+			}
+			if kerrors.IsNotFound(err) {
+				_, err = p.cluster.Clientset.CoreV1().Secrets(newPackageSecret.Namespace).Create(ctx, newPackageSecret, metav1.CreateOptions{})
+				if err != nil {
+					return fmt.Errorf("unable to create the zarf state secret: %w", err)
+				}
+				return nil
+			}
```
[the remark this answers is not in the document]
Comment (phillebaba) [reply src/pkg/packager/remove.go:148]: This should be replaced with SSA TBH but I want to do it in a separate PR for all resources. That should remove any need for create or update.

[the remark this answers is not in the document]
Comment (AustinAbro321) [reply src/pkg/packager/remove.go:148]: By SSA you mean this function call  `c.Clientset.CoreV1().Secrets(cluster.ZarfNamespaceName).Apply` ?

File: src/pkg/packager/remove.go [hunk 2]
```diff
@@ -107,17 +109,49 @@ func (p *Packager) Remove(ctx context.Context) (err error) {
 func (p *Packager) updatePackageSecret(ctx context.Context, deployedPackage types.DeployedPackage) {
 	// Only attempt to update the package secret if we are actually connected to a cluster
 	if p.cluster != nil {
+		// TODO: Error needs to be checked
+		newPackageSecretData, _ := json.Marshal(deployedPackage)
```
Comment (lucasrod16) [inline src/pkg/packager/remove.go:113]: Not sure if you meant to leave this here, but I think we should handle this error and remove the TODO comment

**Review by lucasrod16 — APPROVED** [review]
Left a small nit that doesn't have to be addressed in this PR

File: src/pkg/packager/remove.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (phillebaba) [reply src/pkg/packager/remove.go:148]: Yes

**Comment by phillebaba** [comment]
I have created #2626 to track @AustinAbro321 comments about the verbosity of create or update.

**Review by AustinAbro321 — APPROVED** [review]

**Pull request merged**